### PR TITLE
Fix deserializing `Document`

### PR DESCRIPTION
**Pull request by Mr-Andersen**
When calling `Bot::send_document`, I get an error: `InvalidJson(Error("data did not match any variant of untagged enum TelegramResponse", line: 0, column: 0))`.
Also, when compiling my app with freshly cloned teloxide-core, I got compilation error (see the 1st commit).
This PR fixes both problems.

### CODE REVIEW

File: src/types/document.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ pub struct Document {
     pub file_name: Option<String>,
 
     /// A MIME type of the file as defined by a sender.
-    #[serde(with = "crate::types::non_telegram_types::mime::opt_deser")]
+    #[serde(default, with = "crate::types::non_telegram_types::mime::opt_deser")]
     pub mime_type: Option<Mime>,
```
Comment (WaffleLapkin) [inline src/types/document.rs:33]: I wander if we have the same problems with dates somewhere?

Comment (Mr-Andersen) [reply src/types/document.rs:33]: Likely. I'll search for these cases.

File: src/serde_multipart/mod.rs [hunk 2]
```diff
@@ -31,6 +31,7 @@ pub(crate) fn to_form<T: ?Sized + Serialize>(val: &T) -> impl Future<Output = Re
 }
 
 // https://github.com/teloxide/teloxide/issues/473
+#[cfg(test)]
 #[tokio::test]
```
Comment (WaffleLapkin) [inline src/serde_multipart/mod.rs:35]: Why is this necessary? `#[tokio::test]` [expands to `#[test]`](https://github.com/tokio-rs/tokio/blob/12dd06336d2af8c2d735d4d9e3dc0454ad7942a0/tokio-macros/src/entry.rs#L333) which in turn [applies `cfg(test)`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=931688976a0e166b3f5fc6d2499138a1).

File: src/types/document.rs (hunk 1 above)
Comment (Mr-Andersen) [reply src/types/document.rs:33]: I hope they all match `:[ ]+DateTime`, because I am lazy!

Comment (WaffleLapkin) [reply src/types/document.rs:33]: You can also try searching for `serde(with` or something like that

File: src/serde_multipart/mod.rs (hunk 2 above)
Comment (Mr-Andersen) [reply src/serde_multipart/mod.rs:35]: Weird. I can't reproduce the error, but this is how it looked before (copy-pasted from terminal, true footage):
```
   Compiling teloxide-core v0.3.4 (/home/aka_dude/Work/rust/notify-tg/teloxide-core)
error[E0433]: failed to resolve: could not find `test` in `tokio`
  --> teloxide-core/src/serde_multipart/mod.rs:34:10
   |
34 | #[tokio::test]
   |          ^^^^ could not find `test` in `tokio`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `teloxide-core` due to previous error
```

File: src/types/document.rs (hunk 1 above)
Comment (Mr-Andersen) [reply src/types/document.rs:33]: `:[ ]+Option<DateTime`, I meant. Btw, why does this exist? https://github.com/teloxide/teloxide-core/blob/master/src/types.rs#L248-L250

File: src/serde_multipart/mod.rs (hunk 2 above)
Comment (WaffleLapkin) [reply src/serde_multipart/mod.rs:35]: Oh, that actually makes sense. We only enable tokio macros in `dev` builds, so in other builds the macro can't expand to anything since it doesn't exist
https://github.com/teloxide/teloxide-core/blob/5ccbf8b7b066a1bfde956bfb9d4a6581cd3b5550/Cargo.toml#L49-L51

File: src/types/document.rs (hunk 1 above)
Comment (WaffleLapkin) [reply src/types/document.rs:33]: Because at that time I forgot that `#[serde(default)]` exists and thought that I need a function to do `#[serde(default = "none")]`

Comment (Mr-Andersen) [reply src/types/document.rs:33]: Oh ok

File: src/serde_multipart/mod.rs [hunk 3]
```diff
@@ -31,7 +31,6 @@ pub(crate) fn to_form<T: ?Sized + Serialize>(val: &T) -> impl Future<Output = Re
 }
 
 // https://github.com/teloxide/teloxide/issues/473
-#[cfg(test)]
```
Comment (WaffleLapkin) [inline src/serde_multipart/mod.rs]: `cfg(test)` is actually needed here though https://github.com/teloxide/teloxide-core/pull/159#discussion_r777228274

File: src/types.rs [hunk 4]
```diff
@@ -245,10 +245,6 @@ pub(crate) mod serde_opt_date_from_unix_timestamp {
             .map(|timestamp| DateTime::from_utc(NaiveDateTime::from_timestamp(timestamp, 0), Utc)))
     }
 
-    pub(crate) fn none<T>() -> Option<T> {
-        None
-    }
```
Comment (WaffleLapkin) [inline src/types.rs:250]: `none` is still used in the test down below

File: src/serde_multipart/mod.rs (hunk 2 above)
Comment (Mr-Andersen) [reply src/serde_multipart/mod.rs]: Weird how it works, and compiles most of the time.

File: src/types.rs (hunk 4 above)
Comment (Mr-Andersen) [reply src/types.rs:250]: Fixed

**Comment by WaffleLapkin** [comment]
@Mr-Andersen LGTM. Can you squash the fix commits though, so the git history is cleaner?

**Comment by Mr-Andersen** [comment]
@WaffleLapkin I thought people like small commits...

**Comment by WaffleLapkin** [comment]
@Mr-Andersen we do like small commits, but this branch currently has 2 commits that add `cfg(test)` and one commit that removes it =)

**Comment by Mr-Andersen** [comment]
@WaffleLapkin I think I did that

**Review by WaffleLapkin — APPROVED** [review]
Thanks, good catches!
